app/services/orchestration/task.rb

```diff
@@ -41,5 +44,10 @@ def update_ts
     def <=>(other)
       self.priority <=> other.priority
     end
+
+    def ==(other)
+      return self.id == other.id if self.id && other.id
+      self.name == other.name
```

Comment:
I didn't dig into details yet but my understanding is that we only compare id or name if id is missing, originally we discussed we should also look at all attributes/parameters, is it no longer necessary?

Reply:
I can't compare parameters, there is usually whole object passed in (host, nic) that is modified. It's either ID (recommended) or name (deprecated, will be removed in 1.18).

Reply:
In the next Foreman version, ID will be required. The fallback on name won't be used anymore. We can't easily compare data as it contains ActiveRecord objects.